### ISSUE TITLE
5.12.6 dropped support for 4.4 projection expression

### ISSUE THREAD

**Issue by tomgrossman** [Automattic/mongoose#10188]
critical **bug**

**What is the current behavior?**

mongoose version `5.12.6` dropped support for [MongoDB 4.4 projection expressions](https://docs.mongodb.com/manual/reference/method/db.collection.find/#use-aggregation-expression).
In version `5.12.5` everything works as expected.

Reproduce script:

```javascript
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const dbConnection = require('./conn')

const TestSchema = new Schema({
    field: {
        subField: String
    }
});

TestSchema.statics.getSubField = function () {
    return this.findOne({}, {
        subField: '$field.subField'
    }).lean().exec();
}

async function test () {
    const connection = await dbConnection();
    const TestModel = connection.model('TestModel', TestSchema);
    const newTestModel = new TestModel({field: {subField: 'schema sub field value'}});
    await newTestModel.save();

    const res = await TestModel.getSubField();
    console.log(res.subField);

    await TestModel.deleteMany({}).exec();
}

test();
```

With version `5.12.5` console output:

> /usr/bin/node /home/tom/WebstormProjects/test/model.js
schema sub field value

With version `5.12.6` console output:

> /usr/bin/node /home/tom/WebstormProjects/test/model.js
undefined

**same happens with `find`**

**What is the expected behavior?**
the `projection` in `find` and `findOne` queries, should support aggregation projection expressions, as in `5.12.5` and MongoDB 4.4

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Node.js - 14.16.1
Mongoose - 5.12.6
MongoDB - 4.4

**Comment by Kamikadze4GAME** [comment]
I think this is because of this commit https://github.com/Automattic/mongoose/commit/97f851fb132ee776e73c24fb535f6722e8803838

As far as I understand, this is a temporary workaround to cover up a bug in MongoDB server.  This bug was noticed here https://github.com/Automattic/mongoose/issues/10142

@jwerre reported this bug (https://github.com/Automattic/mongoose/issues/10142#issuecomment-822602279).
The issue had ticket number [SERVER-56130](https://jira.mongodb.org/browse/SERVER-56130), but now it redirects to [SECURITY-721](https://jira.mongodb.org/browse/SECURITY-721) 

So, I think that they will fix it soon.

**Comment by IslandRhythms** [comment]
So on 5.12.5 I also get undefined

```
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const TestSchema = new Schema({
    field: {
        subField: String
    }
});

TestSchema.statics.getSubField = function () {
    return this.findOne({}, {
        subField: '$field.subField'
    }).lean().exec();
}

async function test () {
    const connection = await mongoose.connect('mongodb://localhost:27017/test', {
        useNewUrlParser: true,
        useUnifiedTopology: true
      });
    
    const TestModel = connection.model('TestModel', TestSchema);
    const newTestModel = new TestModel({field: {subField: 'schema sub field value'}});
    await newTestModel.save();

    const res = await TestModel.getSubField();
    console.log(res.subField);

    await TestModel.deleteMany({}).exec();
}

test();
```

**Comment by tomgrossman** [comment]
@IslandRhythms you just added the script, not the result. And you probably did something wrong, it doesn't reproduce on 5.12.5.
Screenshot of running 5.12.6, installing 5 and running with 5:
![image](https://user-images.githubusercontent.com/6324938/117031820-a0432d00-ad09-11eb-9f2d-fe16b06b6e72.png)

**Comment by IslandRhythms** [comment]
@tomgrossman 
![capture](https://user-images.githubusercontent.com/39510674/117034868-b7662880-acd1-11eb-80dc-2ae261f2f6cb.png)

**Comment by tomgrossman** [comment]
@IslandRhythms :thinking: well, that's weird...

**Comment by vkarpov15** [comment]
I took a closer look and you're right @tomgrossman , I'm very sorry for the difficulty. We made this fix in #10142 as a precaution, but you're right that it also hides a useful feature. We will undo this change and release the fix in v5.12.9 :+1:
